site/app/templates/calendar/Calendar.twig

```diff
@@ -86,11 +86,37 @@
         <div class="cal-legend">
             {% if show_all_cookie == 1 %}
                 {% for course in course_names %}
-                    <div name="{{course.course_name|replace({' ':''})}}" class="legend-color"></div>
-                    <p for="{{course.course_name|replace({' ':''})}}-color" class="legend-label">{{course.display_name}}</p>
+                <div 
+                    name="{{ course.course_name|replace({' ':''}) }}" 
```

Comment:
```suggestion
                     id="{{ course.course_name|replace({' ':''}) }}"
                     aria-label="Filter course {{ course.display_name }}"
```
I don't think `name` is valid attribute for div and p elements. 
Also lets add aria-label for better accessibility.